src/consensus/encode.rs

```diff
@@ -1064,5 +1142,21 @@ mod tests {
 
         }
     }
+
+    #[test]
+    fn test_read_bytes_from_finite_reader() {
+        let data : Vec<u8> = (0..10).collect();
+
+        for chunk_size in 1..20 {
+            assert_eq!(
+                &read_bytes_from_finite_reader(
+                    io::Cursor::new(&data),
+                    ReadBytesFromFiniteReaderOpts { len: data.len(), chunk_size }
+                ).unwrap(),
+                &data
```

Comment:
Why references here? This works too
```
            assert_eq!(
                read_bytes_from_finite_reader(
                    io::Cursor::new(&data),
                    ReadBytesFromFiniteReaderOpts { len: data.len(), chunk_size }
                ).unwrap(),
                data
            );
```

Reply:
Natural possessiveness, I guess. :shrug: